packages/@lwc/shared/src/api-version.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import { isNumber } from './language';
+
+export const enum APIVersion {
```

Comment:
In JavaScript, `58 === 58.0` and `59 === 59.0`, so I don't think there's much value in using the decimal here.

In any case, despite the types, you actually can pass in a decimal and it will work. (It will choose the lower value.) There is a Rollup test for this.